File: CodingInterviews/FindContinuousSequence.java
import java.util.ArrayList;

class Solution {
    public int[][] findContinuousSequence(int target) {
        var ans = new ArrayList<int[]>();
        int sum = 0;
        int l = 1, r = 1;

        while (l <= target / 2) {
            if (sum < target)
                sum += r++;
            else if (sum > target)
                sum -= l++;
            else {
                int[] t = new int[r - l];
                for (int k = l; k < r; k++)
                    t[k - l] = k;
                ans.add(t);
                sum -= l++;
            }
        }

        return ans.toArray(new int[ans.size()][]);
    }
}

public class FindContinuousSequence {
    public static void main(String[] args) {
        var sol = new Solution();
        System.out.println(sol.findContinuousSequence(9));
        System.out.println(sol.findContinuousSequence(15));
    }
}

File: CodingInterviews/NowCoder/FindContinuousSequence.cpp
#include "main.h"

class Solution {
  public:
    vector<vector<int>> FindContinuousSequence(int sum) {
        if (sum <= 0)
            return {};
        vector<vector<int>> res;
        int l = 1, r = 2, s = l + r;
        while (sum / 2 + 1 >= r) {
            if (s < sum)
                s += ++r;
            else if (s > sum)
                s -= l++;
            else {
                vector<int> v;
                for (int i = l; i <= r; i++)
                    v.push_back(i);
                res.push_back(v);
                s += ++r;
            }
        }
        return res;
    }
};